tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaHelper.cs

```diff
@@ -179,14 +180,43 @@ internal static Scope CreateProducerScope(Tracer tracer, ITopicPartition topicPa
 
                 if (dataStreamsManager.IsEnabled)
                 {
-                    span.Context.MergePathwayContext(pathwayContext);
+                    // remove any leftover junk they may be left in the headers
+                    message?.Headers?.Remove(DataStreamsPropagationHeaders.TemporaryBase64PathwayContext);
+                    message?.Headers?.Remove(DataStreamsPropagationHeaders.TemporaryEdgeTags);
 
-                    // TODO: we could cache this list in a thread local similar to StringBuilderCache to reduce allocations
-                    var edgeTags = string.IsNullOrEmpty(topic)
-                                       ? new[] { $"group:{groupId}", "type:kafka", }
-                                       : new[] { $"group:{groupId}", $"topic:{topic}", "type:kafka", };
+                    if (!tracer.Settings.KafkaCreateConsumerScopeEnabled && message?.Headers is not null)
```

Comment:
Just recording a thought for later, we might consider an opt-out flag in the future that we could insert here, in case the user has issues with us introducing additional headers on the in-memory object that differs from the actual Kafka message that was on the wire

Reply:
Yeah, that's definitely a possibility. I think it's very unlikely to be an issue, but at least it would be easy to add if necessary